website/docs/sources.md

```diff
@@ -320,10 +320,67 @@ sources:
 
 `gomod` currently does not have any options but may in the future.
 
+
+## Patches
+
+Dalec supports applying patches to sources. Patches must be specified in the
+sources section just like any other type of source.
+To apply a source as a patch to another source there is a patches section that
+is a mapping of the source name you want to apply a patch to, to an ordered list
+of sources that are the patch to apply.
+
+```yaml
+sources:
+  md2man:
+    git:
+      url: https://github.com/cpuguy83/go-md2man.git
+        commit: v2.0.3
+    generate:
+      gomod: {} # Generates a go module cache to cache dependencies
+  md2man-patch:
+    http:
+      url: https://github.com/cpuguy83/go-md2man/commit/fd6bc094ed445b6954a67df55e75d7db95fa8879.patch
+
+patches:
+  m2dman: # Name of the source we want to patch.
+      # Each entry is a patch spec and points to a source listed in the `sources` section
+    - source: md2man-patch # The name of the source that contains the patch
+      path: "" # Path inside the patch source where the patch file is located.
+    # Add more patches to the list (After adding them to the sources section) if needed
+```
+
+Each patch in the list of patch sources MUST be pointing to a file. When the
+patch source is a file-based source, such as `http`, the `path` parameter in the
+patch spec must not be set. When the patch is a directory-based source, such as
+`git`, the `path` parameter in the patch spec must be set AND referecing a file
+in the source.
```

Comment:
Is this correct?
```
When the patch is a directory-based source, such as
`git`, the `path` parameter in the patch spec must be set AND referecing a file
in the source.
```
can you use a git source as the patch? or would that be a `context` source? if both are correct could you also mention `context` as the example below is using a context source?

Reply:
Every directory-based source is supported (context, build, git, image, inline-dir).
Note that all the sources in this doc mention if they are directory or file based.